bblocks/import_tools/imf_weo.py

```diff
@@ -0,0 +1,306 @@
+from dataclasses import dataclass
+import pandas as pd
+from datetime import datetime
+from bs4 import BeautifulSoup
+import io
+import requests
+
+from bblocks.import_tools.common import ImportData, get_response, unzip
+from bblocks.config import BBPaths
+from bblocks.logger import logger
+from bblocks.cleaning_tools import clean
+
+BASE_URL = "https://www.imf.org/"
+
+COLUMN_MAPPER = {
+    "UNIT": "IMF.CL_WEO_UNIT.1.0",
+    "CONCEPT": "IMF.CL_WEO_CONCEPT.1.0",
+    "REF_AREA": "IMF.CL_WEO_REF_AREA.1.0",
+    "FREQ": "IMF.CL_FREQ.1.0",
+    "SCALE": "IMF.CL_WEO_SCALE.1.0",
+}
+
+
+class Parser:
+    """Helper class to parse WEO data
+
+    This class parses the data and schema files into a clean dataframe
+    to be used in the main WEO class
+
+    Parameters:
+        folder: the folder containing the data and schema files
+    """
+
+    def __init__(self, folder: ZipFile):
+        self.folder: ZipFile = folder
+
+        self.data_file: ET = None
+        self.schema_file: ET = None
+        self.data: pd.DataFrame | None = None
+
+    def get_files(self) -> None:
+        """Get the root of files"""
+
+        if len(self.folder.namelist()) > 2:
+            raise ValueError("More than two files in zip file")
+
+        for file in self.folder.namelist():
+            if file.endswith(".xml"):
+                self.data_file = ET.parse(self.folder.open(file)).getroot()
+
+            if file.endswith(".xsd"):
+                self.schema_file = ET.parse(self.folder.open(file)).getroot()
+
+        if self.data_file is None or self.schema_file is None:
+            raise ValueError("No data or schema file found")
+
+    def _extract_data(self) -> None:
+        """extract data from data file"""
+
+        rows = []
+        for series in self.data_file[1].findall("./"):
+            for obs in series.findall("./"):
+                rows.append({**series.attrib, **obs.attrib})
+
+        self.data = pd.DataFrame(rows)
+
+    def _convert_series_codes(self, series: pd.Series, lookup_value: str) -> pd.Series:
+        """Converts a series of codes to the corresponding values in the schema file"""
+
+        query = self.schema_file.findall(
+            "./{http://www.w3.org/2001/XMLSchema}simpleType[@name="
+            + f'"{lookup_value}"'
+            + "]/"
+        )[0].findall("./")
+
+        return series.map({elem.attrib["value"]: elem[0][0].text for elem in query})
+
+    def _clean_data(self) -> None:
+        """Clean and format the dataframe"""
+
+        for col, value in COLUMN_MAPPER.items():
+            self.data = self.data.rename(columns={col: f"{col}_CODE"})
+            self.data[col] = self._convert_series_codes(self.data[f"{col}_CODE"], value)
+
+        self.data = clean.clean_numeric_series(
+            self.data,
+            series_columns=[
+                "REF_AREA_CODE",
+                "LASTACTUALDATE",
+                "TIME_PERIOD",
+                "OBS_VALUE",
+            ],
+        )
```

Comment:
`LASTACTUALDATE`, `TIME_PERIOD` and `REF_AREA_CODE` are integers. You should split this step in two: one where you specify `to=int` for the columns that are integers, and a separate step to clean `OBS_VALUE`, where you can leave blank or specify the default which is `to = float`